compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -693,7 +694,7 @@ abstract class PrepJSInterop extends plugins.PluginComponent
         }
 
         val ownerLoadSpec = jsInterop.jsNativeLoadSpecOf(sym.owner)
-        val jsName = jsInterop.jsNameOf(sym)
+        val JSName.Literal(jsName) = jsInterop.jsNameOf(sym)
```

Comment:
It is not cleaner within the compiler, no. But it avoids complicating the IR, which I think is worth doing.

It is a step back for bincompat, yes. But as long as we don't have a complete solution for perfect backward compat of all `@JSName`, including term members, I'm not sure it's worth fighting for. Also, such a change should only happen if the underlying JS library changes its public API, which is a seriously breaking change to begin with. We can hardly promise more binary compat than the underlying source compat.